frontend/src/lib/util/utils.ts

```diff
@@ -465,3 +466,13 @@ export function isTesting(): boolean {
   })
   return isTesting
 }
+
+export function submitFormWidget(
```

Comment:
This is the method that maybe should not be exposed?

Reply:
Or related questions: since every form is required to have at least one `st.form_submit_button` can't we just do a:

```
if (submitButton !== undefined) {
return
} 
```
within the `widgetMgr.submitForm` implementation?

Reply:
Those are really good points! I'll refactor this method to have each widgetManager to call `submitForm`. I was thinking it was bad to duplicate the code in each widget but it's actually probably reasonable to duplicate it? 

I also think just adding that little snippet in the `submitForm` is also a good solution. I'll check it out today!